OSPSuite.Presentation.Importer/Core/DataSourceFileReaders/ExcelDataSourceFile.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Collections.Generic;
+using OSPSuite.Core.Services;
+using OSPSuite.Infrastructure.Import.Services;
+using OSPSuite.Presentation.Importer.Infrastructure;
+
+namespace OSPSuite.Presentation.Importer.Core.DataSourceFileReaders
+{
+   public interface IExcelDataSourceFile : IDataSourceFile
+   {
+   }
+
+   public class ExcelDataSourceFile : DataSourceFile, IExcelDataSourceFile
+   {
+      public ExcelDataSourceFile(IImportLogger logger) : base(logger)
+      {
+      }
+
+      protected override Dictionary<string, IDataSheet> LoadFromFile(string path)
+      {
+         try
+         {
+            var loadedData = new Dictionary<string, IDataSheet>();
+
+
+            var reader = new ExcelReader(path);
+
+            while (reader.MoveToNextSheet())
+            {
+               var sheetName = reader.CurrentSheet.SheetName;
+               var headers = new List<string>();
+               IDataSheet dataSheet = new DataSheet();
+               dataSheet.RawData = new UnformattedData();
+
+               if (reader.MoveToNextRow())
+                  headers = reader.CurrentRow;
+               //else probably throw exception
+
+               for (var j = 0; j < headers.Count; j++)
+                  dataSheet.RawData.AddColumn(headers[j], j);
+
+               while (reader.MoveToNextRow())
+               {
+                  //the first two could even be done only once
+                  var levels = reader.GetMeasurementLevels(headers.Count);
+                  dataSheet.RawData.CalculateColumnDescription(levels);
+                  dataSheet.RawData.AddRow(reader.CurrentRow);
+               }
+
+               loadedData.Add(sheetName, dataSheet);
+            }
+
+            return loadedData;
+         }
+         catch (Exception ex)
+         {
+            _logger.AddError(ex.ToString());
+            return null;
```

Comment:
null or empty Cache/dictionary?

Reply:
I would say null, since we are not reading an empty file for example. if we reach this, it means something went wrong

Reply:
Sure. But we we do not throw an exception. That means that we need to make sure that the description is explicit about the behavior. 
Also return a Cache instead of dictionary

Reply:
We throw an exception now if something goes wrong

Reply:
new exception created and thrown